### PR TITLE
Add -sil-cross-module-serialize-all option.

### PR DESCRIPTION
**Pull request by dabrahams**
Adds an option to aggressively serialize everything across non-resilient module boundaries, so that non-generic functions don't create an optimization barrier.

Thanks, Michael!

### CODE REVIEW

**Comment by dabrahams** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
To be clear, the point of this option is to enable people to experiment with different inlinable choices.

**Comment by dabrahams** [comment]
@swift-ci please test windows platform

File: lib/SILOptimizer/IPO/CrossModuleSerializationSetup.cpp [hunk 1]
```diff
@@ -181,12 +182,17 @@ makeSubstUsableFromInline(const SubstitutionMap &substs) {
     }
   }
 }
+static llvm::cl::opt<bool> SerializeEverything("sil-cross-module-serialize-all",
+                                               llvm::cl::default(false),
```
Comment (dabrahams) [inline lib/SILOptimizer/IPO/CrossModuleSerializationSetup.cpp:186]: I don't know what I'm doing here, but the Windows compiler very sensibly rejects this use of the `default` keyword.
```suggestion
                                               llvm::cl::init(false),
```

Comment (dabrahams) [inline lib/SILOptimizer/IPO/CrossModuleSerializationSetup.cpp:186]: I don't know what I'm doing here, but the Windows compiler very sensibly rejects this use of the default keyword.
```suggestion
                                               llvm::cl::init(false),
```
@gottesmm, WDYT?

**Comment by dabrahams** [comment]
@swift-ci please test

**Comment by dabrahams** [comment]
@swift-ci please test windows platform

**Comment by gottesmm** [comment]
@dabrahams can you squash this?

**Comment by gottesmm** [comment]
And do a git-clang-format?

**Comment by dabrahams** [comment]
@gottesmm done; git-clang-format invoked without arguments changed nothing, though, FYI.

**Comment by dabrahams** [comment]
n/m I didn't know how to use it correctly; fixed.

**Review by gottesmm — APPROVED** [review]
LGTM! Thanks! Keep me in the loop with what you find!

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Pull request merged**